pkg/kubelet/cm/memorymanager/state/state.go

```diff
@@ -29,16 +29,41 @@ type MemoryTable struct {
 	Free           uint64 `json:"free"`
 }
 
-// MemoryMap contains memory information for each NUMA node.
-type MemoryMap map[int]map[v1.ResourceName]*MemoryTable
-
-// Clone returns a copy of MemoryMap
-func (mm MemoryMap) Clone() MemoryMap {
-	clone := make(MemoryMap)
-	for node, memory := range mm {
-		clone[node] = map[v1.ResourceName]*MemoryTable{}
-		for memoryType, memoryTable := range memory {
-			clone[node][memoryType] = &MemoryTable{
+// NodeState contains NUMA node related information
+type NodeState struct {
+	// NumberOfAssignments contains a number memory assignments from this node
+	// When the container requires memory and hugepages it will increase number of assignments by two
+	NumberOfAssignments int `json:"numberOfAssignments"`
+	// MemoryTable contains NUMA node memory related information
+	MemoryMap map[v1.ResourceName]*MemoryTable `json:"memoryMap"`
+	// NodeGroups contains NUMA nodes that current NUMA node in group with them
+	// It means that we have container that pinned to the current NUMA node and all group nodes
+	Nodes []int `json:"nodes"`
```

Comment:
This is pretty hard to understand based on the description.
`Nodes` contains IDs of NUMA nodes that are in a group with the node that is described by `NodeState`. Is that right?

Reply:
Yes, including that node that described by the `NodeState`
so for example if we have cross NUMA allocation from nodes `[0, 1]`, both nodes will have `[0, 1]` under the `Nodes` field

Reply:
Ok, got it.